docs/data/material/migration/migration-v4/migration-v4.md

```diff
@@ -71,19 +71,36 @@ If you need to support IE 11, check out our [legacy bundle](/material-ui/guides/
 
 ## Update React & TypeScript version
 
+### Update React
+
 The minimum supported version of React has been increased from v16.8.0 to v17.0.0.
```

Comment:
**Off-topic**

@mui/core I almost wonder if we shouldn't have done #25464. Most seems to keep the support of React 16.8.0

- https://github.com/ant-design/ant-design/blob/c2f75a2ba3054f67041810aa8d172ab5965b21b6/package.json#L309
- https://github.com/react-bootstrap/react-bootstrap/blob/bd016aa330b005073f79ec1439ad3fef5eb021b4/package.json#L150
- https://github.com/radix-ui/primitives/blob/f0a7495bcd00faa14164b2f80721149238c85d9d/packages/react/context-menu/package.json#L28
- https://github.com/adobe/react-spectrum/blob/810579b671791f1593108f62cdc1893de3a220e3/packages/react-aria/package.json#L63

Reply:
Honestly, if we support 17, I don't see why we shouldn't support 16.8. But we would need to probably add additional tests for it. I wonder if people still use it honestly

Reply:
> I wonder if people still use it

@mnajdova based on https://www.npmjs.com/package/react-dom?activeTab=versions we could cover an extra 7797292 downloads of React DOM (downloads >=v16.8.0 && < v17.0.0. So with Material UI v5, we excluded 45% of the React DOM downloads.

> But we would need to probably add additional tests for it.

We have tests for React 17 in https://github.com/mui/material-ui/blob/90d6a3bffd33472a371eaa4d29b75751d282b282/.circleci/config.yml#L832 which have been broken for at least 3 months https://app.circleci.com/insights/github/mui/material-ui/workflows/react-17/overview?branch=master&reporting-window=last-90-days

Reply:
> Which have been broken for at least 3 months https://app.circleci.com/insights/github/mui/material-ui/workflows/react-17/overview?branch=master&reporting-window=last-90-days

Probably this means that it was a bad decision on my side to create this check as "non blocking" on the PR CI. 

> @mnajdova based on https://www.npmjs.com/package/react-dom?activeTab=versions I could cover an extra 7797292 weekly downloads of React DOM (downloads >=v16.8.0 && < v17.0.0). So with Material UI v5, we exclude 45% of the today's React DOM downloads.

I am not sure that this is that much black/white. If someone is not aiming to update React, it's likely that they are not going to update their UI library dependency either, so they may be depending on v4. But agree, it is a big percentage, so it may be worth checking what would take for us to support v16.8.0.

Reply:
> Probably this means that it was a bad decision on my side to create this check as "non blocking" on the PR CI.

@mnajdova I think that a nice workaround could be to run this step randomly on PRs. This way, we get the feedback every now and then, without slowing down the CI too much.

> so they may be depending on v4.

This could be why v4 is still this high in downloads, too many have to first upgrade React, but they might not because some other libraries are not compatible with React 17 yet. I have created a live app with Toolpad to better visualize v16.8 use, with live data: https://master--toolpad.mui.com/deploy/clcdobvof0008n99jpx86fede/pages/331kqzd

<img width="237" alt="Screenshot 2023-01-01 at 19 57 07" src="https://user-images.githubusercontent.com/3165635/210181803-482e43e5-345a-45da-9cc1-982107c8fc0f.png">

It seems to have been down during the holidays period.

Reply:
Ok, let's bring this up with the team.